Essentials/src/main/java/com/earth2me/essentials/config/EssentialsConfiguration.java

```diff
@@ -0,0 +1,432 @@
+        final CommentedConfigurationNode node = configurationNode.node(toSplitRoot(path));
+        if (node.virtual()) {
+            return null;
+        }
+        return node;
+    }
+
+    public boolean hasProperty(final String path) {
+        return !configurationNode.node(toSplitRoot(path)).virtual();
+    }
+
+    public Object[] toSplitRoot(String node) {
+        node = node.startsWith(".") ? node.substring(1) : node;
+        return node.contains(".") ? node.split("\\.") : new Object[]{node};
+    }
+
+    public synchronized void load() {
+        if (pendingWrites.get() != 0) {
+            LOGGER.log(Level.INFO, "Parsing config file {0} has been aborted due to {1} current pending write(s).", new Object[]{configFile, pendingWrites.get()});
+            return;
+        }
+
+        if (configFile.getParentFile() != null && !configFile.getParentFile().exists()) {
+            if (!configFile.getParentFile().mkdirs()) {
+                LOGGER.log(Level.SEVERE, tl("failedToCreateConfig", configFile.toString()));
+                return;
+            }
+        }
+
+        if (!configFile.exists()) {
+            if (legacyFileExists()) {
+                convertLegacyFile();
+            } else if (altFileExists()) {
+                convertAltFile();
+            } else if (templateName != null) {
+                try (final InputStream is = resourceClass.getResourceAsStream(templateName)) {
+                    LOGGER.log(Level.INFO, tl("creatingConfigFromTemplate", configFile.toString()));
+                    Files.copy(is, configFile.toPath());
+                } catch (IOException e) {
+                    LOGGER.log(Level.SEVERE, tl("failedToWriteConfig", configFile.toString()), e);
+                }
+            }
+        }
+
+        try {
+            configurationNode = loader.load();
+        } catch (final ParsingException e) {
+            final File broken = new File(configFile.getAbsolutePath() + ".broken." + System.currentTimeMillis());
+            if (configFile.renameTo(broken)) {
+                LOGGER.log(Level.SEVERE, "The file " + configFile.toString() + " is broken, it has been renamed to " + broken.toString(), e.getCause());
+                return;
+            }
+            LOGGER.log(Level.SEVERE, "The file " + configFile.toString() + " is broken. A backup file has failed to be created", e.getCause());
+        } catch (final ConfigurateException e) {
+            LOGGER.log(Level.SEVERE, e.getMessage(), e);
+        } finally {
+            // Something is wrong! We need a node! I hope the backup worked!
+            if (configurationNode == null) {
+                configurationNode = loader.createNode();
+            }
+        }
+    }
+
+    public boolean legacyFileExists() {
+        return false;
+    }
+
+    public void convertLegacyFile() {
+
+    }
+
+    public boolean altFileExists() {
+        return false;
+    }
+
+    public void convertAltFile() {
+
+    }
```

Comment:
Not sure what this is meant for, but looks like it should either be implemented or removed

Reply:
They are implemented in EssentialsUserConfiguration but are noops for regular configs :P